docs/cri/config.md

```diff
@@ -200,6 +200,11 @@ version = 2
       # config files being loaded from the CNI config directory.
       cni_max_conf_num = 1
 
+      # sandbox_sizing_from_pause_container allows specifying the sandbox resource requirements via the pause container
+      # when creating a sandbox using the task service API. Only a runtime that expects to consume this sizing information
+      # and still constrain pause container appropriately should set this to true.
+      sandbox_sizing_from_pause_container = false
```

Comment:
I'd rather not expose the internal implementation detail for pod implementation. Also not convinced yet that the requirement for handling pod overall resources & pod overhead should be done on a per runtime configuration switch basis. The incoming field from the run pod request should be enough to "switch" on handling where possible, and if not we should at least log a warning.

It's fair to say a non vm runc type implementation for this will be different than a vm shim, and we'll probably want to handle that in sandbox down the road I think.

Let's have a chat with @dmcgowan this week..pls.

Reply:
If kubelet only adds these fields on container runtime handlers they expect this to work on ... then we could reject the request till we have an implementation for a particular type of runtime.

tempted to say something like just add an annotation like we did for a few other sandbox fields.. but will reserve saying how we should do this till we can chat about it :-)

Reply:
nod, saw it is currently setting on all, was just pointing out "if" kubelet did it another way we could reject (fail the request) till we had support in the runtime type for the optionally set field (the switch). Also noting that using annotation/label pattern wouldn't be a change in behavior unless the annotation was processed accordingly. Further noting that even if/when this was implemented for a runc type .. I don't see us implementing it with a switch for changing the pause container resource spec :-)